server/src/main/java/com/cloud/usage/UsageServiceImpl.java

```diff
@@ -216,6 +216,31 @@ public boolean generateUsageRecords(GenerateUsageRecordsCmd cmd) {
             s_logger.debug("Account details not available. Using userContext accountId: " + accountId);
         }
 
+        // Check if a domain admin is allowed to access the requested account info.
+        if (_accountService.isDomainAdmin(caller.getId()) && accountId != null){
```

Comment:
@Spaceman1984 What is the caller account is root admin?

Reply:
or normal user account?

Reply:
A normal user doesn't have access to this API, this check is only applicable to domain admin.